packages/forms/src/Commands/Concerns/CanGenerateForms.php

```diff
@@ -89,40 +87,40 @@ protected function getResourceFormSchema(string $model): string
                 $componentData['image'] = [];
             }
 
-            if ($column->getNotnull()) {
+            if (! $column['nullable']) {
                 $componentData['required'] = [];
             }
 
-            if (in_array($column->getType()::class, [
-                Types\BigIntType::class,
-                Types\DecimalType::class,
-                Types\FloatType::class,
-                Types\IntegerType::class,
-                Types\SmallIntType::class,
+            if (in_array($type['name'], [
+                'integer',
+                'decimal',
+                'float',
+                'double',
+                'money',
             ])) {
                 if ($componentData['type'] === Forms\Components\TextInput::class) {
                     $componentData['numeric'] = [];
                 }
 
-                if (filled($column->getDefault())) {
-                    $componentData['default'] = [$column->getDefault()];
+                if (filled($column['default'])) {
+                    $componentData['default'] = [$column['default']];
```

Comment:
Doctrine DBAL doesn't support column's **default with expression** but Laravel does. Therefore, Doctrine always parses the default expression to get a literal value, but we don't do that on Laravel - returning the default expression itself. The best we can do here is to parse default expression manually. for example:

```php
$table->integer('foo')->default(23);
$table->string('bar')->default('hello');
```

`$column['default ']` for `foo` would be:
* MySQL: `23`
* PostgreSQL: `'23'::smallint`
* SQL Server: `((23))`
* SQLite: `'23'`

`$column['default ']` for `bar` would be:
* MySQL: `hello`
* MariaDB: `'hello'`
* PostgreSQL: `'hello'`
* SQL Server: `('hello')`
* SQLite: `'hello'`

Reply:
Fixed at 4295f0d